### DISCUSSION TITLE
Blurred raster tiles VTM vs. Mapsforge

### DISCUSSION THREAD

**Discussion by demiantres** [mapsforge/vtm#1027] [Q&A] [answered]
Raster tiles (such as OpenStreetMap tiles) at the same zoom level are rendered sharper with the Mapsforge engine than with the VTM engine:

Mapsforge:
![image](https://user-images.githubusercontent.com/5980561/229293299-056fcbad-d5fb-44ab-ac22-716a5df5421e.png)

VTM:
![image](https://user-images.githubusercontent.com/5980561/229293329-775310a7-9da7-4f03-bc6c-122bf3ef529b.png)

What could be the reason for this? OpenGL texture compression?

**Comment by devemux86** [comment, marked as answer]
(besides possible OpenGL reasons)

VTM has fractional zoom, it works with absolute scale (double), the zoom is a derived  value.
(see `MapPosition` for the math)

The displayed VTM zoom 14.0  may not be the exact Mapsforge zoom 14 (absolute integer).

**Reply by demiantres** [reply]
Understood. But when I blend Mapsforge and VTM map screenshots in Photoshop, VTM is clearly blurred even with exactly the same element size (for instance the label "Ostwald" is exactly the same size). Even with minimal fractional zoom steps I never get a sharp image in VTM.

**Reply by demiantres** [reply]
When you look at the grid coordinates in the two screenshots above then we see the same problem there as well.
